scripts/open-coredump.sh

```diff
@@ -0,0 +1,130 @@
+
+set -e
+
+SCYLLA_S3_RELOC_SERVER_URL="http://backtrace.scylladb.com"
+
+if [[ $# != 1 ]]; then
+	echo Please provide the path to the coredump
+	exit 1
+fi
+
+for required in eu-unstrip jq curl git; do
+	if ! type $required >& /dev/null; then
+		echo Please install $required first
+		exit 1
+	fi
+done
+
+COREFILE=$1
+
+BUILD_ID=$(eu-unstrip -n --core ${COREFILE} | cut -f2 -d' ' | cut -f1 -d@)
+
+echo "Build id: ${BUILD_ID}"
+
+BUILD=$(curl -s -X GET "${SCYLLA_S3_RELOC_SERVER_URL}/build.json?build_id=${BUILD_ID}")
+
+RESPONSE_BUILD_ID=$(jq -r .build_id <<< $BUILD)
+VERSION=$(jq -r .version <<< $BUILD)
+PRODUCT=$(jq -r .product <<< $BUILD)
+RELEASE=$(jq -r .release <<< $BUILD)
+ARCH=$(jq -r .arch <<< $BUILD)
+BUILD_MODE=$(jq -r .build_mode <<< $BUILD)
+PACKAGE_URL=$(jq -r .package_url <<< $BUILD)
+
+if [[ "$RESPONSE_BUILD_ID" != "$BUILD_ID" ]]
+then
+    echo "Mismatching build id: requested ${BUILD_ID} but got ${RESPONSE_BUILD_ID}";
+    exit 1
+fi
+
+echo "Matching build is ${PRODUCT}-${VERSION} ${RELEASE} ${BUILD_MODE}-${ARCH}"
+
+PACKAGE_FILE=$(basename ${PACKAGE_URL})
+
+if ! [[ -d ${PRODUCT} ]]
+then
+    if ! [[ -f ${PACKAGE_FILE} ]]
+    then
+        echo "Downloading relocatable package from ${PACKAGE_URL}"
+        curl --output ./${PACKAGE_FILE} ${PACKAGE_URL}
+    else
+        echo "Relocatable package ${PACKAGE_URL} already downloaded"
+    fi
+
+    echo "Extracting package ${PACKAGE_FILE}"
+    tar -zxf $PACKAGE_FILE
+else
+    echo "Relocatable package ${PACKAGE_URL} already downloaded and extracted"
+fi
+
+COMMIT_HASH=$(cut -f3 -d. <<< $RELEASE)
+BASE_VERSION=$(grep -o "^[0-9]\+\.[0-9]\+" <<< $VERSION)
+BRANCH=branch-${BASE_VERSION}
+
+if ! [[ -d ${PRODUCT}.git ]]
+then
+    echo "Cloning ${PRODUCT}.git"
+    git clone -b ${BRANCH} https://github.com/scylladb/${PRODUCT}.git ${PRODUCT}.git
+    cd ${PRODUCT}.git
+    git checkout -q ${COMMIT_HASH}
+    # Skip the other submodules, they are not needed for debugging
+    git submodule update -q --depth=1 --init seastar
+    cd ..
+else
+    echo "${PRODUCT}.git already cloned"
+fi
+
+if ! [[ -f scylla-gdb.py ]]
+then
+    MAIN_BRANCH=master
+    if [[ "${PRODUCT}" == "scylla-enterprise" ]]
+    then
+        MAIN_BRANCH=enterprise
+    fi
+
+    echo "Downloading scylla-gdb.py"
+    curl --output scylla-gdb.py https://raw.githubusercontent.com/scylladb/${PRODUCT}/${MAIN_BRANCH}/scylla-gdb.py
+else
+    echo "scylla-gdb.py already present, not downloading"
+fi
+
+cat << EOF
+Launching dbuild container.
+
+To examine the coredump with gdb:
+
+    $ gdb -q -ex 'set directories scylla.git' --core ${COREFILE} /opt/scylladb/libexec/scylla
+
+Load the scylla-gdb.py utility script with:
+
+    (gdb) source scylla-gdb.py
+
+See https://github.com/scylladb/scylladb/blob/master/docs/dev/debugging.md for more information on how to debug scylla.
+
+Good luck!
+EOF
+
+exec ./${PRODUCT}.git/tools/toolchain/dbuild -it -v $(pwd):/workdir -v $(pwd)/scylla:/opt/scylladb -w /workdir -- bash -l
```

Comment:
Sine its running a container with gdb, maybe it makes sense to provide .gdbinit with known useful quirks. Like libpthread-debug=1 and solib-search-path (but presumably with deps installed this one is not needed indeed)

Reply:
I think it would be better to allow the user to provide their own `.gdbinit`. Currently, no extra flags or options are required for debugging to work correctly. I don't use any `.gdbinit` the vast majority of the time.